src/Gui/Tree.cpp

```diff
@@ -540,6 +540,31 @@ void TreeWidgetItemDelegate::initStyleOption(QStyleOptionViewItem *option,
     }
 }
 
+class DynamicQLineEdit : public ExpLineEdit
+{
+public:
+    DynamicQLineEdit(QWidget *parent = nullptr) : ExpLineEdit(parent) {}
+
+    QSize sizeHint() const override
+    {
+        QSize size = QLineEdit::sizeHint(); 
+        QFontMetrics fm(font());
+        int availableWidth = parentWidget()->width() - geometry().x() - 20; // Calculate available width
+        size.setWidth(std::min(fm.horizontalAdvance(text()) + 30, availableWidth)); // Add some padding and max width

```

Comment:
Where are `20` and `30` come from?

Reply:
Some padding as to not overflow on the right, but might not be necessary.
I'll remove it and test.

Reply:
I changed the 30 to : 
```c++
2 * (style()->pixelMetric(QStyle::PM_FocusFrameHMargin) + 1)
                    + 2 * style()->pixelMetric(QStyle::PM_LayoutHorizontalSpacing)
                    + TreeParams::getItemBackgroundPadding();
```
I took inspiration `QRect TreeWidgetItemDelegate::calculateItemRect()`, but I am not certain of what I did.

Reply:
Not really as this is something that was mostly a result of trial and error. How it really should look like:

```c++
style()->pixelMetric(QStyle::PM_PaddingLeft) + style()->pixelMetric(QStyle::PM_PaddingRight) + 
style()->pixelMetric(QStyle::PM_BorderLeft) + style()->pixelMetric(QStyle::PM_BorderRight) + 
TreeParams::getItemBackgroundPadding();
```

However, Qt does not expose ability to read padding or border. From my experiments I concluded that to achieve somewhat reliable spacing default horizontal margins of QFrames combined with default layout horizontal padding can be used to have good enough results.